docs/06-using-cairo.md

```diff
@@ -0,0 +1,160 @@
+prone. It is the programmer responsability to properly check return and errors
+values.
+
+## Using LGI in Awesome
+
+GObject and Gnome centric libraries tend to use the common C practive of
+emulating namespaces using underscores in funciton name. LGI expose a proper
+namespace based API. For example, if the C function is:
+
+    cairo_image_surface_create()
+
+Then LGI equivalent is:
+
+    lgi.cairo.ImageSurface.create()
+
+The same goes for enums:
+
+    CAIRO_FORMAT_ARGB32
+
+become:
+
+    lgi.cairo.Format.ARGB32
+
+LGI is also object oriented while the C API is function based. When those
+function take the "class" "object", then this:
+
+    cairo_line_to(cr, x, y)
+
+can be expressed as:
+
+    cr:line_to(x, y)
+
+It is however important to note some inconsistencies. For example, Cairo is
+called `lgi.cairo` while GLib is called `lgi.GLib`. Figuring this out will
+require some tried and errors. The best way to do this without actually
+reloading Awesome is to open the `lua` command in a terminal and using `print`:
+
+    print("This will print a table address:", require("lgi").cairo)
+    print("This will print an error:", require("lgi").Cairo)
+
+It is recommanded to avoid using `require` everytime you use a function and
+instead include the libraries at the top of your `rc.lua` or lua module
+
+    local cairo = require("lgi").cairo
+
+## The Cairo API
+
+Cairo is a 2D graphic library used by Awesome, Gnome and XFCE. It allows to
+paint paths on a `surface`. Awesome use it internally and being able to call
+is directly should be considered a power user feature.
+
+Understanding these few important concepts is necessary to be able to use Cairo:
+
+**Surface**:
+
+A surface is the area where the painting will be done. There is multiple types
+of surfaces including:
+
+ * Color pixmap with transparency (`ARGB32`) or without (`RGB24`)
+ * Monochrome pixmap surfaces with transparency (`A8`) or without (`A1`)
+ * SVG vectorial surfaces
+ * Native (XCB) surfaces
+ * Framebuffers and other less interesting ones from an AwesomeWM point of view.
+
+For more details, read:
+
+* [Surfaces](https://cairographics.org/manual/cairo-Image-Surfaces.html).
+
+**Sources**:
+
+Sources are elements like colors, patterns or gradient. See `gears.color` for
+common sources.
+
+For more details, read:
+
+* [Pattern](http://cairographics.org/manual/cairo-cairo-pattern-t.html)
+
+**Context and paths**:
+
+A context is the proxy between the program and the surface. The context hold a
+set of paths. Paths are something like a line, circle or rectangle. They may or
+may not be closed (a shape). All operations on the surfaces has to be told to
+the context. Until the context state is flushed (see next section), those paths
+add up, until that, nothing will be written to the surface. For example:
+
+    cr:rectangle(0 , 0 , 10, 10)
+    cr:rectangle(10, 10, 10, 10)
+
+wont do anything until and operation is applied on the context.
```

Comment:
instead of "the operation", how about "a drawing operation" or "a drawing operation using a path"?
And instead of "is applied to the context", how about "is done"?